master/internal/db/postgres_trial.go

```diff
@@ -625,6 +619,24 @@ WHERE id = $1;
 			}
 		}
 		return nil
+	}
+
+	return rollbacks, run(tx)
+}
+
+// addTrialMetrics inserts a set of trial metrics to the database.
+func (db *PgDB) addTrialMetrics(
+	ctx context.Context, m *trialv1.TrialMetrics, isValidation bool,
+) (rollbacks map[string]int, err error) {
+	switch v := m.Metrics.AvgMetrics.Fields["epoch"].AsInterface().(type) {
+	case float64, nil:
+	default:
+		return nil, fmt.Errorf("cannot add metric with non numeric 'epoch' value got %v", v)
+	}
+	rollbacks = make(map[string]int)
```

Comment:
I guess this does not have to be initialized because it'll be overwritten anyway?
```suggestion
	var rollbacks map[string]int
```